src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,160 @@
+package duke;
+
+import exceptions.FileUnloadableException;
+import exceptions.ParseTaskFromStringException;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.ToDo;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * The Storage class handles reading and writing tasks to/from a file.
+ */
+public class Storage {
+    private static String filePath = "tasks.txt";
+
+    /**
+     * Constructs a Storage instance with the specified file path.
+     *
+     * @param filePath The path to the file storing tasks.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Converts a date string to a LocalDateTime object.
+     *
+     * @param by The date string to be converted.
+     * @return The LocalDateTime object representing the date.
+     * @throws DateTimeParseException If the date string is in an invalid format.
+     */
+    public static LocalDateTime saveAsDate(String by) throws DateTimeParseException {
+        try {
+            // Try to parse the input as "yyyy-MM-dd" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        } catch (DateTimeParseException e) {
+            by += " 0000";
+            // If parsing as "yyyy-MM-dd" format fails, try to parse as "yyyy-MM-dd HHmm" format
+            return LocalDateTime.parse(by, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+        }
+    }
```

Comment:
I agree with what HollaG has mentioned, saveAsDate doesn't really capture the essence of converting a string into another form.